src/Munkres.jl

```diff
@@ -98,6 +100,38 @@ function munkres(costMat::AbstractMatrix{T}) where {T<:Real}
     return Zs
 end
 
+function munkres(costMat::AbstractMatrix) 
+    # check the type of the matrix (cannot do it within the function prototype, 
+    # as Matrix{Float64} <: (Matrix{T} where T <: Real) is false)
+    if typeof(eltype(costMat)) == Union
+        T = eltype(costMat).a <: Real ? eltype(costMat).a : eltype(costMat).b
+    else
+        T = eltype(costMat)
+    end
+    T <: Real || throw(TypeError(:munkres, "", AbstractMatrix{Union{Missing, T}} where T <: Real, costMat))
```

Comment:
I think you could use triangular dispatch `foo(x::AbstractArray{S}) where {T<:Real,S<:Union{Missing,T}}` to get `T`. Here is a toy example:
```
julia> foo(x::AbstractArray{T}) where {T<:Real} = "real"
foo (generic function with 1 method)

julia> foo(x::AbstractArray{S}) where {T<:Real,S<:Union{Missing,T}} = "missing or real", T
foo (generic function with 2 methods)

julia> foo([missing,2,3])
("missing or real", Int64)

julia> foo([1,2,3])
"real"
```